### PR TITLE
[dagster-dbt] Log a deprecation warning for users using log_column_level_metadata

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Logs a new warning message telling users who are using the log_column_level_metadata macro that it will be deprecated in 1.9, and pointing them to fetch_column_metadata instead.

## Test Plan

Unit test.

### CODE REVIEW

**Comment by benpankow** [comment]
* **#23068** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23068?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @benpankow and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by rexledesma — DISMISSED** [review]
Can this occur on `DbtCliResource(...)` instantiation time instead? We should just emit this warning if the dagster package is installed in their dbt deps

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_metadata/packages.yml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 # We keep use `packages.yml` for compatability with `dbt-core==1.5.*`.
 # Once we remove support for that version, we should rename this file to `dependencies.yml`
 packages:
-  - local: "../../../dbt_packages/dagster"
+  - git: "https://github.com/dagster-io/dagster.git"
```
Comment (benpankow) [inline python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_metadata/packages.yml:4]: Cut over to published version so we can remove this package now (or in the future) and the warning test will still work.

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_metadata/packages.yml [hunk 2]
```diff
@@ -1,4 +1,6 @@
...
+    subdirectory: "python_modules/libraries/dagster-dbt/dbt_packages/dagster"
+    revision: 1.7.13
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_metadata/packages.yml:6]:
```suggestion
    revision: master
```

File: python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py [hunk 3]
```diff
@@ -1825,6 +1837,20 @@ def my_dbt_op(dbt: DbtCliResource):
...
+        packages_yaml = Path(self.project_dir).joinpath("packages.yml")
+        dependencies_yaml = Path(self.project_dir).joinpath("dependencies.yml")
+
+        if context.log and (
+            (packages_yaml.exists() and _dbt_packages_has_dagster_dbt(packages_yaml))
+            or (dependencies_yaml.exists() and _dbt_packages_has_dagster_dbt(dependencies_yaml))
+        ):
+            context.log.warn(
+                "Fetching column metadata using `log_column_level_metadata` macro is deprecated and will be"
+                " removed in Dagster 1.9. Use the `fetch_column_metadata` method in your asset definition"
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py:1850]: Use the `dagster-dbt` version

**Review by rexledesma — APPROVED** [review]

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_metadata/packages.yml (hunk 2 above)
Comment (benpankow) [reply python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_metadata/packages.yml:6]: Ah yeah, I guess we can lock to a tag if we remove it 👍

**Pull request merged**